sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala

```diff
@@ -131,7 +131,7 @@ trait HashJoin extends JoinCodegenSupport {
   @transient protected lazy val streamedBoundKeys =
     bindReferences(HashJoin.rewriteKeyExpr(streamedKeys), streamedOutput)
 
-  protected def buildSideKeyGenerator(): Projection =
+  protected def buildSideKeyGenerator(): UnsafeProjection =
```

Comment:
keep consistent w/ `def streamSideKeyGenerator()`, actually this method can be removed since it's unused